### PR TITLE
Set default offset to an Offset object

### LINKED ISSUE #174: `TensorDataset.offset` is set to an `int` but `Dataset.offset` is set to an `Offset`
By default, offset in the tensor dataset class is set to 1, rather than an offset value. 

this raises an error when I don't define an offset when constructing a tensordataset.

```
File ~/CEBRA-dev/cebra/solver/base.py:430, in Solver.fit(self, loader, valid_loader, save_frequency, valid_frequency, decode, logdir, save_hook)
    427 self.to(loader.device)
    429 iterator = self._get_loader(loader)
--> 430 for num_steps, batch in iterator:
    431     self.model.train()
    432     stats = self.step(batch)

File ~/CEBRA-dev/cebra/solver/util.py:96, in ProgressBar.__iter__(self)
     94 if self.use_tqdm:
     95     self.iterator = tqdm.tqdm(self.iterator)
---> 96 for num_batch, batch in enumerate(self.iterator):
     97     yield num_batch, batch

File ~/miniconda3/envs/nlb/lib/python3.12/site-packages/tqdm/std.py:1181, in tqdm.__iter__(self)
   1178 time = self._time
   1180 try:
-> 1181     for obj in iterable:
   1182         yield obj
   1183         # Update and possibly print the progressbar.
   1184         # Note: does not call self.update(1) for speed optimisation.

File ~/CEBRA-dev/cebra/data/base.py:262, in Loader.__iter__(self)
    260 for _ in range(len(self)):
    261     index = self.get_indices(num_samples=self.batch_size)
--> 262     yield self.dataset.load_batch(index)

File ~/CEBRA-dev/cebra/data/multi_session.py:103, in MultiSessionDataset.load_batch(self, index)
     99 def load_batch(self, index: BatchIndex) -> List[Batch]:
    100     \"\"\"Return the data at the specified index location.\"\"\"
    101     return [
    102         cebra_data.Batch(
--> 103             reference=session[index.reference[session_id]],
    104             positive=session[index.positive[session_id]],
    105             negative=session[index.negative[session_id]],
    106             index=index.index,
    107             index_reversed=index.index_reversed,
    108         ) for session_id, session in enumerate(self.iter_sessions())
    109     ]

File ~/CEBRA-dev/cebra/data/single_session.py:96, in SingleSessionDataset.__getitem__(self, index)
     88 def __getitem__(self, index: Iterable) -> npt.NDArray[np.float64]:
     89     \"\"\"
     90     Args:
     91         index: List of index to return from the data.
   (...)
     94         [ No.Samples x Neurons x 10 ]
     95     \"\"\"
---> 96     index = self.expand_index(index).to(self.device)
     97     return self.neural[index].transpose(2, 1)

File ~/CEBRA-dev/cebra/data/base.py:141, in Dataset.expand_index(self, index)
    123 \"\"\"
    124 
    125 Args:
   (...)
    136     Requires the :py:attr:`offset` to be set.
    137 \"\"\"
    139 # TODO(stes) potential room for speed improvements by pre-allocating these tensors/
    140 # using non_blocking copy operation.
--> 141 offset = torch.arange(-self.offset.left,
    142                       self.offset.right,
    143                       device=index.device)
    145 index = torch.clamp(index, self.offset.left,
    146                     len(self) - self.offset.right)
    148 return index[:, None] + offset[None, :]

AttributeError: 'int' object has no attribute 'left'"
}
```

Potential solutions: 
- change int to offset similar to the dataset class where: ` self.offset: Offset = cebra.data.Offset(0, 1)`.
- keep it as an arg, but offset.
- keep it as an arg, int, and create the offset from the int.

### LINKED ISSUE #175: pytorch not recognized when building the docs
pytorch not recognized when building the docs in the git actions. 

`WARNING: py:class reference target not found: torch.Tensor`

see https://github.com/AdaptiveMotorControlLab/CEBRA/actions/runs/11036962413/job/30656677245?pr=166 for example.

### PR DESCRIPTION
**Pull request by icarosadero**
Should fix #174

### CODE REVIEW

**Comment by icarosadero** [comment]
@cla-bot check

**Review by stes — CHANGES_REQUESTED** [review]
@icarosadero i think the offset here is actually not used, right? We could also consider deprecating the argument instead.

Do you have a code example that's fixed by this change?

File: cebra/data/datasets.py [hunk 1]
```diff
@@ -71,7 +72,7 @@ def __init__(self,
                  neural: Union[torch.Tensor, npt.NDArray],
                  continuous: Union[torch.Tensor, npt.NDArray] = None,
                  discrete: Union[torch.Tensor, npt.NDArray] = None,
-                 offset: int = 1,
+                 offset: Offset = Offset(1),
```
Comment (stes) [inline cebra/data/datasets.py:75]:
```suggestion
                 offset: Offset = None,
```

**Review by stes — COMMENTED** [review]
set to `None` (unused)

File: cebra/data/datasets.py (hunk 1 above)
Comment (stes) [inline cebra/data/datasets.py:75]:
```suggestion
                 offset: Offset = Offset(0, 1),
```

otherwise, a symmetric offset is chosen

**Review by stes — APPROVED** [review]

**Comment by stes** [comment]
can be merged once tests pass

**Review by stes — CHANGES_REQUESTED** [review]
@icarosadero if you get the chance, could you add a test that reproduces this error https://github.com/AdaptiveMotorControlLab/CEBRA/issues/174#issue-2533087585 to this PR?

**Review by stes — APPROVED** [review]
Thanks for your contribution, @icarosadero , lgtm!

**Pull request merged**